### PR TITLE
ARACHNE-2446. Make update of DataNode partial

### PR DESCRIPTION
**Pull request by aklochkova**
now DataNode may be updated partially

### CODE REVIEW

File: src/main/java/com/odysseusinc/arachne/portal/api/v1/controller/BaseDataNodeController.java [hunk 1]
```diff
@@ -159,16 +159,28 @@ protected DataNode buildEmptyDataNode() {
...
             Principal principal
     ) throws PermissionDeniedException, NotExistException, AlreadyExistException, BindException, ValidationException {
 
-        validateOrganization(commonDataNodeRegisterDTO.getOrganization());
-
+        //for the first DN update all fields (name, description, organization) are mandatory in commonDataNodeRegisterDTO.
+        // In further updates they may be empty and will be taken from existing record
+        DN existingDN = baseDataNodeService.getById(dataNodeId);
+        if (existingDN.getName() == null) {
+            validateDNFields(commonDataNodeRegisterDTO);
```
Comment (pavgra) [inline src/main/java/com/odysseusinc/arachne/portal/api/v1/controller/BaseDataNodeController.java]: Why not `validator.validate`?

File: src/main/java/com/odysseusinc/arachne/portal/api/v1/controller/BaseDataNodeController.java [hunk 2]
```diff
@@ -159,16 +159,28 @@ protected DataNode buildEmptyDataNode() {
...
         final DN dataNode = conversionService.convert(commonDataNodeRegisterDTO, getDataNodeDNClass());
         dataNode.setId(dataNodeId);
-        final Organization organization = conversionService.convert(commonDataNodeRegisterDTO.getOrganization(), Organization.class);
+        final Organization organization;
+        if (commonDataNodeRegisterDTO.getOrganization() != null) {
+            organization = conversionService.convert(commonDataNodeRegisterDTO.getOrganization(), Organization.class);
+        } else {
+            organization = existingDN.getOrganization();
+        }
+        if (commonDataNodeRegisterDTO.getDescription() == null) {
```
Comment (pavgra) [inline src/main/java/com/odysseusinc/arachne/portal/api/v1/controller/BaseDataNodeController.java]: make checks consistently (in prev place you are comparing for not null, here for null). I'd suggest to pick `existingDn` and replace values in it (for newer ones) when it is applicable

File: src/main/java/com/odysseusinc/arachne/portal/service/OrganizationService.java [hunk 3]
```diff
@@ -29,6 +29,8 @@
 public interface OrganizationService {
     Organization create(Organization organization) throws ValidationException;
 
+    Organization get(String name);
```
Comment (agackovka) [inline src/main/java/com/odysseusinc/arachne/portal/service/OrganizationService.java]: getByName? findByName?

Comment (aklochkova) [reply src/main/java/com/odysseusinc/arachne/portal/service/OrganizationService.java]: fixed

**Pull request merged**